test_main.go

```diff
@@ -6,32 +6,40 @@ import (
 	"testing"
 
 	"github.com/matrix-org/complement/b"
+	"github.com/matrix-org/complement/config"
 	"github.com/matrix-org/complement/ct"
 )
 
 var (
 	testPackage    *TestPackage
-	customDeployer func(numServers int) Deployment
+	customDeployer func(t ct.TestLike, numServers int, config *config.Complement) Deployment
 )
 
 type complementOpts struct {
-	cleanup          func()
-	customDeployment func(numServers int) Deployment
+	// Args:
+	// - We pass in the Complement config (`testPackage.Config`) so the deployer can inspect
+	// `DebugLoggingEnabled`, `SpawnHSTimeout`, `PackageNamespace` etc.
+	cleanup func(config *config.Complement)
+	// Args:
+	// - We pass in `t` as there needs to be a way to handle an error in the custom deployer.
+	// - We pass in the Complement config (`testPackage.Config`) so the deployer can inspect
+	// `DebugLoggingEnabled`, `SpawnHSTimeout`, `PackageNamespace`, etc.
```

Comment:
Another use case for needing `config.Complement`  is needing access to `CACertificate`/`CAPrivateKey` since they won't magically be in your environment (normally mounted under `/complement/ca/ca.crt`, etc)